src/integrationTest/java/com/mongodb/hibernate/query/select/SimpleSelectQueryIntegrationTests.java

```diff
@@ -0,0 +1,186 @@
+import jakarta.persistence.Table;
+import java.util.List;
+import java.util.function.Consumer;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.query.SelectionQuery;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.jspecify.annotations.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(annotatedClasses = SimpleSelectQueryIntegrationTests.Contact.class)
+@ExtendWith(MongoExtension.class)
+class SimpleSelectQueryIntegrationTests {
+
+    @BeforeEach
+    void beforeEach(SessionFactoryScope scope) {
+        scope.inTransaction(session -> List.of(
+                        new Contact(1, "Bob", 18, Country.USA),
+                        new Contact(2, "Mary", 35, Country.CANADA),
+                        new Contact(3, "Dylan", 7, Country.CANADA),
+                        new Contact(4, "Lucy", 78, Country.CANADA),
+                        new Contact(5, "John", 25, Country.USA))
+                .forEach(session::persist));
+    }
+
+    @Test
+    void testComparisonByEQ(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country = :country",
+                q -> q.setParameter("country", Country.USA.name()),
+                List.of(1, 5)));
+    }
+
+    @Test
+    void testComparisonByNE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session,
+                "from Contact where country != ?1",
+                q -> q.setParameter(1, Country.USA.name()),
+                List.of(2, 3, 4)));
+    }
+
+    @Test
+    void testComparisonByLT(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age < :age", q -> q.setParameter("age", 35), List.of(1, 3, 5)));
+    }
+
+    @Test
+    void testComparisonByLTE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age <= ?1", q -> q.setParameter(1, 35), List.of(1, 2, 3, 5)));
+    }
+
+    @Test
+    void testComparisonByGT(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age > :age", q -> q.setParameter("age", 18), List.of(2, 4, 5)));
+    }
+
+    @Test
+    void testComparisonByGTE(SessionFactoryScope scope) {
+        scope.inTransaction(session -> assertContactQueryResult(
+                session, "from Contact where age >= :age", q -> q.setParameter("age", 18), List.of(1, 2, 4, 5)));
+    }
+
+    @Test
+    void andFilterTest(SessionFactoryScope scope) {
```

Comment:
(consistency) Let's start all tests with `test`.

Reply:
We are already not doing that in the codebase. But if we are to come up with a rule, maybe, given that "test" in the name is noise that makes names longer, we consider the style where "test" is never specified?

Reply:
currently multiple testing method naming conventions have co-existed in the repo already. I didn't object to dropping "test" from method names in @stIncMale 's testing code.

Reply:
but I agree, in this PR, there is no reason not to change `xxxTest` to `testxxx`. Updated.

Reply:
Going forward, we will prefix all tests with the verb `test`, and the rest of the name should tersely make clear what is being tested.